data/transactions/logic/assembler_test.go

```diff
@@ -592,6 +612,178 @@ func TestAssembleBytesString(t *testing.T) {
 	}
 }
 
+func TestAssembleOptimizedConstants(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Bytes", func(t *testing.T) {
+		t.Parallel()
+
+		program := `
+byte 0x0102
+byte base64(AQI=) // 0x0102
+byte base32(AEBA====) // 0x0102
+byte "test"
+byte base32(ORSXG5A=) // "test"
+addr WSJHNPJ6YCLX5K4GUMQ4ISPK3ABMS3AL3F6CSVQTCUI5F4I65PWEMCWT3M
+byte 0x0103
+byte base64(AQM=) // 0x0103
+byte base32(AEBQ====) // 0x0103
+`
+		// 0x0102 and 0x0103 are tied for most frequent int, but 0x0102 should win because it appears first
+		expected := `
+bytecblock 0x0102 0x0103 0x74657374
+bytec_0 // 0x0102
+bytec_0 // 0x0102
+bytec_0 // 0x0102
+bytec_2 // "test"
+bytec_2 // "test"
+pushbytes 0xb49276bd3ec0977eab86a321c449ead802c96c0bd97c2956131511d2f11eebec // addr WSJHNPJ6YCLX5K4GUMQ4ISPK3ABMS3AL3F6CSVQTCUI5F4I65PWEMCWT3M
+bytec_1 // 0x0103
+bytec_1 // 0x0103
+bytec_1 // 0x0103
+`
+		for v := uint64(backBranchEnabledVersion); v <= AssemblerMaxVersion; v++ {
+			t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
+				expectedOps, err := AssembleStringWithVersion(expected, v)
```

Comment:
consider testProg here (anywhere you just need to assemble correctly).  It only saves one line (the error check), but when it fails, it reports the problem in more detail.